### PR TITLE
Fix: Report generation in cli-dashboard

### LINKED ISSUE #283: Huge discrepancy between CLI generated report and exported CSV
**Describe the bug**
Huge discrepancy between CLI generated report and exported CSV

**To Reproduce**
1. Generate report (websites that reproduced the issue on my end: https://www.mercadopago.com.ar, https://www.rakuten.co.jp) using [CLI](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/A.-PSAT's-How-To#build-and-run-psats-cli)
2. Open the report
3. Export the report via the export button on the report
4. Import the following reports on Google Sheet (or any spreadsheet) : `affected-cookies.csv`, `cookies.csv`, `report.csv`
5. Find the discrepancies in the # of cookies below

**Observed discrepancies**
1. Discrepancy in the total # of cookies: localhost report: **21** (1P cookies x 16 + 3P cookies x 5), exported report: **79** (1P x 16 + 3P x 63)
2. Discrepancy in the # of affected cookies: localhost report: **3**, exported report: **19**

**Expected behavior**
The # of cookies in the localhost report and exported report should be the same

**Screenshots**
1. localhost report > total cookies list screenshot
Screenshot 2023-11-20 at 11 42 00

2. exported report > total cookies list screenshot
Screenshot 2023-11-20 at 11 58 54

3.  localhost report > affected cookie list screenshot
Screenshot 2023-11-20 at 12 09 25

4. exported report > affected cookie list screenshot
Screenshot 2023-11-20 at 12 07 33

**Desktop (please complete the following information):**
 - OS: Macbook Pro Intel, Sonoma 14.1
 - PSAT version : [v0.3.2](https://github.com/GoogleChromeLabs/ps-analysis-tool/releases/tag/v0.3.2)

**Additional context**
- Also a minor bug that total # of cookies in the exported report > `report.csv` is wrong (it says 81 but expecting 79)
- The issue does not seem to reproduce for sites with smaller number of cookies.

### LINKED ISSUE #314: [QA] CLI Dashboard Shows wrong count for total cookies in report.csv file
<kbd>image</kbd>

-----

> [!Note]
The code is pulled from the dev branch

### PR DESCRIPTION
**Pull request by ayushnirwal**
## Description

As mentioned in issues #314 and #283 there was a mismatch between the exported report (`report.csv`) and the one shown in the CLI dashboard. This was due to miscounting cookies used ( or set/updated ) in requests generated from multiple frames. 

This PR aims to fix that miscounting error and reactors the report generation routine into multiple routines one for each file to make testing easier. 

## Relevant Technical Choices

- Reactors [`reportDownloader.ts`](https://github.com/GoogleChromeLabs/ps-analysis-tool/blob/develop/packages/cli-dashboard/src/components/utils/reportDownloader.ts) into multiple routines in the folder [`generateReports`](https://github.com/GoogleChromeLabs/ps-analysis-tool/tree/fix/report-csv-data/packages/cli-dashboard/src/components/utils/generateReports).
- Reimplements summary data file (`report.csv`) generation in the function [`generateSummaryDataCSV`](https://github.com/GoogleChromeLabs/ps-analysis-tool/blob/fix/report-csv-data/packages/cli-dashboard/src/components/utils/generateReports/generateSummaryDataCSV.ts)

## Screenshot/Screencast
Screenshot 2023-12-04 at 11 47 38 AM
Screenshot 2023-12-04 at 11 45 26 AM

Fixes #314
Fixes #283

### CODE REVIEW

File: packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * External dependencies
+ */
+import { sanitizeCsvRecord } from '@ps-analysis-tool/common';
+/**
+ * Internal dependencies
+ */
+import type { CompleteJson, CookieJsonDataType } from '../../../types';
+
+export const affectedCookieDataHeader = [
```
Comment (mayan-000) [inline packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts:26]: How about if it's a constant we can use capitalized snack case?

File: packages/cli-dashboard/src/components/utils/generateReports/generateAllCookiesCSV.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * External dependencies
+ */
+import { sanitizeCsvRecord } from '@ps-analysis-tool/common';
+/**
+ * Internal dependencies
+ */
+import type { CompleteJson, CookieJsonDataType } from '../../../types';
+
+export const cookieDataHeader = [
```
Comment (mayan-000) [inline packages/cli-dashboard/src/components/utils/generateReports/generateAllCookiesCSV.ts:26]: Capitalized Snack Case.

File: packages/cli-dashboard/src/components/utils/generateReports/generateSummaryDataCSV.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  const cookieMap: Map<string, CookieJsonDataType> = new Map();
+
+  // More than one frame can use one cookie, need to make a map for gettig unique entries.
+  Object.entries(frameCookieDataMap).forEach(([, { frameCookies }]) => {
+    Object.entries(frameCookies).forEach(([cookieKey, cookieData]) => {
+      cookieMap.set(cookieKey, cookieData);
+    });
+  });
+
+  let totalCookies = 0;
```
Comment (mayan-000) [inline packages/cli-dashboard/src/components/utils/generateReports/generateSummaryDataCSV.ts:34]: Query: Can `map.size` work for assigning totalCookies?

File: packages/cli-dashboard/src/components/utils/generateReports/generateTechnologyCSV.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * External dependencies
+ */
+import { sanitizeCsvRecord } from '@ps-analysis-tool/common';
+/**
+ * Internal dependencies
+ */
+import type { CompleteJson } from '../../../types';
+
+const technologyDataHeader = [
```
Comment (mayan-000) [inline packages/cli-dashboard/src/components/utils/generateReports/generateTechnologyCSV.ts:26]: Capitalized snack case

File: packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts (hunk 1 above)
Comment (amovar18) [reply packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts:26]: Second this idea,  but do we need to export this? I don't see any reference to this variable outside this file.

File: packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  'Category',
+  'GDPRPortal',
+];
+
+const generateAffectedCookiesCSV = (siteAnalysisData: CompleteJson): string => {
+  const frameCookieDataMap = siteAnalysisData.cookieData;
+
+  const affectedCookieMap: Map<string, CookieJsonDataType> = new Map();
+
+  // More than one frame can use one cookie, need to make a map for gettig unique entries.
```
Comment (amovar18) [inline packages/cli-dashboard/src/components/utils/generateReports/generateAffectedCookiesCSV.ts:46]: Nitpick can ignore this `getting` spelling is wrong.

File: packages/cli-dashboard/src/components/utils/generateReports/generateAllCookiesCSV.ts (hunk 2 above)
Comment (amovar18) [reply packages/cli-dashboard/src/components/utils/generateReports/generateAllCookiesCSV.ts:26]: Second this idea,  but do we need to export this? I don't see any reference to this variable outside this file.

**Review by amovar18 — CHANGES_REQUESTED** [review]

**Review by mayan-000 — APPROVED** [review]
Small changes requested.

LGTM.

**Pull request merged**